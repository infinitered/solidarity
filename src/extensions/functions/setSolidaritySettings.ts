import { SolidarityRunContext, SolidaritySettings } from '../../types'
module.exports = (settings: SolidaritySettings, context: SolidarityRunContext): void => {
  const { print, filesystem } = context

  try {
    if (settings.requirements) {
      // Write file
      filesystem.write('.solidarity', JSON.stringify(settings, null, 2), {atomic: true})
    } else {
      throw 'You must have a requirements key to be a valid solidarity file'
    }
  } catch (e) {
    print.error(e)
    process.exit(1)
  }
}
