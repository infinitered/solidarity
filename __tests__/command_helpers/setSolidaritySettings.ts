import setSolidaritySettings from '../../src/extensions/functions/setSolidaritySettings'

test('setSolidaritySettings exists', () => expect(setSolidaritySettings).toMatchSnapshot())
