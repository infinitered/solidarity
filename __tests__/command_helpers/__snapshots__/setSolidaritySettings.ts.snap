// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setSolidaritySettings exists 1`] = `[Function]`;
